src/components/auth/HandleCallback.tsx

```diff
@@ -1,20 +1,52 @@
-import React from 'react';
-import { useLocation } from 'react-router-dom';
+import React, { useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
 
-import OidcCallback from './OidcCallback';
 import { isCallbackUrl } from '../../auth/index';
-import { getClient } from '../../auth/oidc-react';
 import { ROUTES } from '../../enums';
+import { LoginCallbackHandler, OidcClientError, useApiTokensClient, useOidcClient, User } from 'hds-react';
 
 const HandleCallback = (props: React.PropsWithChildren<unknown>): React.ReactElement => {
   const { children } = props;
   const location = useLocation();
-  const client = getClient();
-  const authenticated = client.isAuthenticated();
+  const navigate = useNavigate();
+  const { isAuthenticated } = useOidcClient();
+  const [userOrError, setUserOrError] = useState<User | OidcClientError | undefined>(undefined);
   const isCallBack = isCallbackUrl(location.pathname);
+  const { fetch } = useApiTokensClient();
 
-  if (!authenticated && isCallBack) {
-    return <OidcCallback successRedirect={ROUTES.INDEX} failureRedirect={ROUTES.AUTH_ERROR} />;
+  const onSuccess = (user: User) => {
+    setUserOrError(user);
+    // Following makes sure api tokens are available
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const tokens = fetch(user);
+    navigate(ROUTES.INDEX, { replace: true });
+  };
+
+  const onError = (error: OidcClientError | undefined) => {
+    // eslint-disable-next-line no-console
+    console.error(error);
+    setUserOrError(error);
+    if (!error) return;
+    if (
+      error.isSignInError &&
+      error.message === 'Current state (HANDLING_LOGIN_CALLBACK) cannot be handled by a callback'
```

Comment:
Removed commented line. Did some refactoring and uncommented a line that should've been present in order to avoid having that "Login failed" -page appear.